### PR TITLE
Improve documentation for DSL attribute hash keys [RFC] [Guides]

### PR DESCRIPTION
**Pull request by Ashton-W**
I want to improve the Guides for the DSLs. Currently they don't list the acceptable hash keys, there are only examples from which you can gleam this important information.
https://guides.cocoapods.org/syntax/podspec.html#source

I'm trying to find a good pattern to get this information into the documentation system, then we will also have to update **guides.cocoapods.org** to format and include it. Right now I'm just using `yard` to see how it picks up the comments.

Idealy, we shouldn't have to repeat ourselves. Where we define the options should be where the documentation is. I'm finding it hard to do this, and I'm not happy with the `@overload` usage here.

:bird: @orta

### CODE REVIEW

File: lib/cocoapods-core/specification/dsl.rb [hunk 1]
```diff
@@ -290,8 +290,36 @@ module DSL
...
       #                     :sha1 => '7e21857fe11a511f472cfd7cfa2d979bd7ab7d96' }
       #
-      #   @param  [Hash{Symbol=>String}] source
-      #           The location from where the library should be retrieved.
+      #
+      #   @overload source=(git)
+      #     @param  [Hash{Symbol=>String}] git
+      #     @option git [String] :git URI where library git repo can be cloned.
+      #     @option git [String] :tag git tag
+      #     @option git [true or false] :submodules checkout submodules
```
Comment (segiddins) [inline lib/cocoapods-core/specification/dsl.rb]: `Boolean`?

**Comment by segiddins** [comment]
I like this! I personally feel overload is pretty spot on here.

**Comment by Ashton-W** [comment]
I'll continue on this path then, unless someone has a better idea of where and how to put the comments for the keys.

**Comment by Ashton-W** [comment]
Requesting more feedback.

There is more room for improvement in other areas where we rely on examples, but I'll get to those in other PRs.

Now to try and get it in **guides**.

**Comment by orta** [comment]
don't worry about getting it in the guides, every CP release it will be automatically migrated in,

**Comment by orta** [comment]
Overall looks good to me, more docs is :+1:

**Comment by neonichu** [comment]
@Ashton-W good work :+1:

**Comment by Ashton-W** [comment]
@orta guides doesn't surface these particular documentation tags yet

**Comment by segiddins** [comment]
@Ashton-W because we haven't cut a new release of CocoaPods yet ;)

**Comment by Ashton-W** [comment]
I'm running guides locally with the Core gem pointing to my fork. @overloads and @options don't show up.
Am I missing something?

**Comment by segiddins** [comment]
You've updated the Core submodule in `gems` ?

**Comment by Ashton-W** [comment]
Yes. Other changes are reflected in the generated site:

image

**Comment by segiddins** [comment]
OK, thanks for opening the issue over there, I'll try and take a look before I release the 0.39 beta

**Pull request merged**